### ISSUE TITLE
Completion

### ISSUE THREAD

**Issue by Alizter** [ejgallego/coq-lsp#50]
Code completion gives suggestions about what the user might want to type. We need to investigate groups of objects and priorities.

**Comment by ejgallego** [comment]
Yup, there is already basic support in #41 , but needs more work before merge, and upstream thanks to https://github.com/coq/coq/pull/8766

Some more concrete TODOs here:
- we need to refine how to select the completion token from a position. Here we have two cases:
  + the ast for the position at point is available, then we can actually build an interval map and use a fancy heuristic.
  + parsing failed and only a text span is available. Maybe we can still call Coq's tokenizer on that span, but OMMV.
- completion of notations need to build a (cached) prefix map
- map objects to their kinds
